CIP-????/README.md

```diff
@@ -0,0 +1,394 @@
+-   The current metadata for a token can be deterministically reconstructed by starting from the latest CIP-25 update to the token, and applying the subsequent CIP-???? metadata updates in ascending blockchain transaction order.
+
+Both CIP-25 and CIP-???? updates affect the token metadata:
+
+-   A CIP-25 update removes all fields in the current metadata state and replaces them with a new complete definition of the metadata state.
+-   A CIP-???? update selectively adds, removes, or modifies fields relative to the current metadata state. It does not apply to any asset class that has not had tokens previously minted or any asset class that has not had CIP-25 metadata previously defined, before the CIP-???? update.
+
+However, we recommend that CIP-???? adopters use the new metadata update mechanism exclusively to manage all updates to their token metadata after the initial CIP-25 metadata is set.
+
+Our proposal only affects the `????` top-level CBOR tag of the metadata field in Cardano transactions. (Note: we’re using `????` as a stand-in for the CIP number that will eventually be assigned to this proposal.)
+
+## Assigning a metadata oracle for a token policy ID
+
+A token metadata oracle is defined via two addresses:
+
+-   An oracle update address that is authorized to post metadata updates. For example, this address could be controlled by a bot that automatically posts token metadata updates as needed to the blockchain.
+-   An oracle main address that is used to update the oracle update address. This address should be controlled via private keys held in cold storage or hardware wallets.
+
+A metadata oracle can be explicitly assigned to a policy ID by setting the following metadata in a transaction that mints or burns tokens for that policy:
+
+```json
+{
+    "????": {
+        "assign_metadata_oracle": {
+            "<policyId>": {
+                "main_address": "<Shelley_address>",
+                "update_address": "<Shelley_address>"
+            }
+        }
+    }
+}
+```
+
+While a metadata oracle is not explicitly assigned to a native-script-based policy ID, the policy ID is implicitly assigned a metadata oracle with both addresses set to an address derived from the native script. Specifically, they are both set to an Enterprise address constructed by setting the payment key to the verification key from the native script and keeping the staking key empty.
+
+## Updating an oracle assignment
+
+The metadata oracle assignment for a policy ID can be updated via a transaction signed by the oracle main address key. The transaction must only send ADA from the oracle main address to itself and must not mint any tokens, but it may contain any number of inputs and outputs. Otherwise, the transaction is ignored for the purposes of metadata oracle assignment.
+
+The schema for updating an oracle assignment is the same as for the initial assignment in the minting transaction:
+
+```json
+{
+    "????": {
```

Comment:
Can you add cddl definitions for all of these? Since those are what are actually stored on the ledger

Reply:
Good idea! Thanks

Reply:
Yes, certainly. We'll add the CDDL definitions for everything, this week.

Reply:
@GeorgeFlerovsky I am working on it, btw. :D

Reply:
@sebastiengllmt @smaugpool
We've added CDDL files to our proposal.

Similar to CIP-25, we have:

- `cddl/version_1.cddl` for text-based policy IDs and asset names.
- `cddl/version_2.cddl` for bytes-based policy IDs and asset names.

Reply:
@SebastienGllmt we had added the definition files